File: class/class10.1/Team.cpp
#include "Team.hpp"

Team::Team(const std::string& name) : name(name) {}

void Team::updateStats(int scored, int conceded, int yellow, int red, double maxHeight) {
    goalsScored += scored;
    goalsConceded += conceded;
    yellowCards += yellow;
    redCards += red;
    maxBallHeight += maxHeight;
}

void Team::addPoints(int points) {
    pointsAccumulated += points;
}

int Team::goalDifference() const {
    return goalsScored - goalsConceded;
}

std::string Team::getName() const {
    return name;
}

double Team::getPoints() const {
    return pointsAccumulated;
}

double Team::getMaxBallHeight() const {
    return maxBallHeight;
}

int Team::getGoalsScored() const {
    return goalsScored;
}

int Team::getGoalsConceded() const {
    return goalsConceded;
}

int Team::getYellowCards() const {
    return yellowCards;
}

int Team::getRedCards() const {
    return redCards;
}

File: home/home10.1/Team.swift
import Foundation

class Team {
    let name: String
    private(set) var goalsScored = 0
    private(set) var goalsConceded = 0
    private(set) var yellowCards = 0
    private(set) var redCards = 0
    private(set) var maxBallHeight: Double = 0.0
    private(set) var pointsAccumulated = 0.0

    init(name: String) {
        self.name = name
    }

    func updateStats(scored: Int, conceded: Int, yellow: Int, red: Int, maxHeight: Double) {
        goalsScored += scored
        goalsConceded += conceded
        yellowCards += yellow
        redCards += red
        maxBallHeight += maxHeight
    }

    func addPoints(_ points: Int) {
        pointsAccumulated += Double(points)
    }

    func goalDifference() -> Int {
        return goalsScored - goalsConceded
    }

    func isCountry404() -> Bool {
        return name.lowercased() == "russia"
    }
}
